### PR TITLE
Meal plan rework

### PR DESCRIPTION
**Pull request by ArminShinobi**
- meal_plan_overview has a new background image and now loads the next meal when you press on reroll
- visualize nutrient now has also the same background and css is adjusted
- new db entries in user to save the current meal nr

### CODE REVIEW

File: meal_plan_overview/meal_plan_overview.php [hunk 1]
```diff
@@ -35,46 +47,88 @@
...
+        <div class="spinner-grow text-primary" role="status" style="width: 3rem; height: 3rem; z-index: 20;">
+        </div>
+    </div>
+</div>
 <main class="d-flex align-items-center justify-content-center card-design">
-    <div>
+    <div id="anchor" style="display:none">
         <div class="d-flex align-items-center justify-content-center">
-            <h1 class="m-3" style="color:white;">Meal plan overview</h1>
+            <h1 class="m-3">Meal plan overview</h1>
```
Comment (Zylesto) [inline meal_plan_overview/meal_plan_overview.php:59]: I would suggest Meal Plan Overview. It looks better and will blend better in with the rest if the website

File: meal_plan_overview/recipes.php [hunk 2]
```diff
@@ -6,36 +6,63 @@
...
+    if (isset(json_decode($request)->function_name)) {
+        $function_name = json_decode($request)->function_name;
+        if ($function_name == "get_recipes_by_user_id") {
+            $result = get_recipes_by_user_id($conn, json_decode($request)->userId);
+        } else if ($function_name == "get_taste_and_nutrient_visualization") {
+            $result = get_taste_and_nutrient_visualization(json_decode($request)->meal_id);
+        } else if ($function_name == "increment_current_meal_nr") {
+            $result = increment_current_meal_nr($conn, json_decode($request)->userId, json_decode($request)->meal_type_nr);
+        } else {
+            $result["errorText"] = "Can't call function: " . $function_name;
```
Comment (Zylesto) [inline meal_plan_overview/recipes.php:18]: I would add an dot to the end of the error text, to be consistent

File: meal_plan_overview/recipes.php [hunk 3]
```diff
@@ -6,36 +6,63 @@
...
+        }
+    } else {
+        $result["errorText"] = "Function_name is not set or is empty";
```
Comment (Zylesto) [inline meal_plan_overview/recipes.php:21]: I would add an dot to the end of the error text, to be consistent

File: meal_plan_overview/recipes.php [hunk 4]
```diff
@@ -6,36 +6,63 @@
...
-        echo json_encode(["recipes" => $recipes, "error" => false, "errorText" => ""]);
+
+    if (isset($result["errorText"])) {
+        echo json_encode(["result" => null, "error" => true, "errorText" => $result["errorText"]]);
+    } else {
+        echo json_encode(["result" => $result, "error" => false, "errorText" => ""], JSON_INVALID_UTF8_IGNORE);
     }
 } else {
-    echo json_encode(["recipes" => null, "error" => true, "errorText" => "Request is not set or is empty"]);
+    echo json_encode(["result" => null, "error" => true, "errorText" => "Request is not set or is empty"]);
```
Comment (Zylesto) [inline meal_plan_overview/recipes.php:30]: I would add an dot to the end of the error text, to be consistent

File: meal_plan_overview/taste_and_nutrition_visualization.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+          integrity="sha384-BmbxuPwQa2lc/FVzBcNJ7UAyJxM6wuqIj61tLrc4wSX0szH/Ev+nYRRuWlolflfl" crossorigin="anonymous"/>
+    <link rel="stylesheet" href="../shared/css/shared_overlay.css"/>
+    <link rel="stylesheet" href="../shared/css/shared_background2.css"/>
+    <link rel="stylesheet" href="taste_and_nutrition_visualization.css"/>
+    <script type="module">
+        import {tasteAndNutrientVisualization} from "./get_recipes.js";
+        window.tasteAndNutrientVisualization = tasteAndNutrientVisualization;
+    </script>
+
+    <title>Taste and nutrition visualization</title>
```
Comment (Zylesto) [inline meal_plan_overview/taste_and_nutrition_visualization.html:19]: Maybe call it "Taste And Nutrition Visualization", to be more consistent

**Review by Zylesto — APPROVED** [review]

**Review by ordass00 — APPROVED** [review]

**Pull request merged**